src/stores/utils/gas.js

```diff
@@ -0,0 +1,32 @@
+const { toWei } = require('web3').utils
+
+export async function fetchGasPrice({ bridgeContract, oracleFn }) {
+  let gasPrice = null
+  try {
+    gasPrice = await oracleFn()
+  } catch (e) {
+    if (!e.message.includes('Gas Price Oracle url not defined')) {
+      console.error(`Gas Price API is not available. ${e.message}`)
+    }
+
+    try {
+      gasPrice = await bridgeContract.methods.gasPrice().call()
```

Comment:
When I ran the bridge UI, I discovered that it is not correct to get fallback gas price for the contract. It is assumed that this gas price needs to be used by the bridge validator rather than the user. That's why gas prices for these two roles could differ (e.g. the validator will be able to send transactions with zero gas price where as the user will not be able to do so).
Let's have the following logic: request the gas price from the oracle and if it cannot be received, use the fallback gas price from the environment variable.

Reply:
Done